Don't show error messages for prismjs missing languages

## 💥 Proposal

Currently, it's possible to use `usePrism: true` to set prismjs as the syntax highlighter for all languages. However when a language that prismjs doesn't support is used, the errors are logged to console. Personally think this could be done much better.

```
 Error: Cannot find module 'prismjs/components/prism-html.min'
    at Function.Module._resolveFilename (module.js:548:15)
    at Function.Module._load (module.js:475:25)
    at Module.require (module.js:597:17)
    at require (internal/module.js:11:18)
    at Function.require (/Users/kyleroach/Developer/react-native-elements/website/node_modules/docusaurus/lib/core/renderMarkdown.js:52:17)
    at Object.rules.fence (/Users/kyleroach/Developer/react-native-elements/website/node_modules/remarkable/lib/rules.js:73:37)
    at Renderer.render (/Users/kyleroach/Developer/react-native-elements/website/node_modules/remarkable/lib/renderer.js:71:39)
    at Remarkable.render (/Users/kyleroach/Developer/react-native-elements/website/node_modules/remarkable/lib/index.js:152:24)
    at MarkdownRenderer.render (/Users/kyleroach/Developer/react-native-elements/website/node_modules/docusaurus/lib/core/renderMarkdown.js:103:26)
    at toHtml (/Users/kyleroach/Developer/react-native-elements/website/node_modules/docusaurus/lib/core/renderMarkdown.js:113:43) code: 'MODULE_NOT_FOUND' }
```

References https://github.com/facebook/Docusaurus/pull/735#issuecomment-395484508

### Have you read the [Contributing Guidelines on issues](https://github.com/facebook/Docusaurus/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

--- a/v1/lib/core/renderMarkdown.js
+++ b/v1/lib/core/renderMarkdown.js
@@ -56,7 +56,9 @@ class MarkdownRenderer {
               } catch (err) {
                 if (err.code === 'MODULE_NOT_FOUND') {
                   const unsupportedLanguageError = chalk.yellow(
-                    `Warning: language-${language} is not supported by prismjs.` +
+                    `Warning: ${chalk.red(
+                      language,
+                    )} is not supported by prismjs.` +
                       '\nPlease refer to https://prismjs.com/#languages-list for the list of supported languages.',
                   );
                   console.log(unsupportedLanguageError);

--- a/v1/lib/core/renderMarkdown.js
+++ b/v1/lib/core/renderMarkdown.js
@@ -9,19 +9,21 @@ const hljs = require('highlight.js');
 const Markdown = require('remarkable');
 const prismjs = require('prismjs');
 const deepmerge = require('deepmerge');
-
+const chalk = require('chalk');
 const anchors = require('./anchors.js');
 
 const CWD = process.cwd();
 
 const alias = {
   js: 'jsx',
+  html: 'markup',
+  sh: 'bash',
+  md: 'markdown',
 };
 
 class MarkdownRenderer {
   constructor() {
     const siteConfig = require(`${CWD}/siteConfig.js`);
-
     let markdownOptions = {
       // Highlight.js expects hljs css classes on the code element.
       // This results in <pre><code class="hljs css languages-jsx">
@@ -45,14 +47,20 @@ class MarkdownRenderer {
                 siteConfig.usePrism.length > 0 &&
                 siteConfig.usePrism.indexOf(lang) !== -1)
             ) {
+              const language = alias[lang] || lang;
               try {
-                const language = alias[lang] || lang;
                 // Currently people using prismjs on Node have to individually require()
                 // every single language (https://github.com/PrismJS/prism/issues/593)
                 require(`prismjs/components/prism-${language}.min`);
                 return prismjs.highlight(str, prismjs.languages[language]);
               } catch (err) {
-                console.error(err);
+                if (err.code === 'MODULE_NOT_FOUND') {
+                  const unsupportedLanguageError = chalk.yellow(
+                    `Warning: language-${language} is not supported by prismjs.` +
+                      '\nPlease refer to https://prismjs.com/#languages-list for the list of supported languages.',
+                  );
+                  console.log(unsupportedLanguageError);
+                } else console.error(err);
               }
             }
             if (hljs.getLanguage(lang)) {

